absl/meta/type_traits.h

```diff
@@ -278,6 +278,10 @@ using remove_extent_t = typename std::remove_extent<T>::type;
 template <typename T>
 using remove_all_extents_t = typename std::remove_all_extents<T>::type;
 
+#ifdef __GNUC__
```

Comment:
We have a macro to do this correctly that should be used here:
https://github.com/abseil/abseil-cpp/blob/66406fdf1553de6ad672576167e7cc7ca2d764cb/absl/base/attributes.h#L688-L708

I *think* the `__clang__` case is for `clang-cl`.